scripts/arm_build.sh

```diff
@@ -0,0 +1,84 @@
+#! /usr/bin/bash
+
+echo "Mandel ARM Build"
+
+if [[ $NAME != "Ubuntu" ]]
+   then
+      echo "Currently only supporting Ubuntu based builds. Proceed at your own risk."
+fi
+
+if [ $# -eq 0 ] || [ -z "$1" ]
+   then
+      echo "Please supply a directory for the mandel build and a value for the number of jobs to use for building."
+      echo "The binary packages will be created and placed into the mandel build directory."
+      echo "./pinned_build.sh <mandel build dir> <1-100>"
+      exit -1
+fi
+
+CORE_SYM=EOS
+MANDEL_DIR=$1
+JOBS=$2
+SCRIPT_DIR="$( cd -- "$( dirname -- "${BASH_SOURCE[0]:-$0}"; )" &> /dev/null && pwd 2> /dev/null; )";
+
+
+pushdir() {
+   DIR=$1
+   mkdir -p ${DIR}
+   pushd ${DIR} &> /dev/null
+}
+
+popdir() {
+   EXPECTED=$1
+   D=`popd`
+   popd &> /dev/null
+   echo ${D}
+   D=`eval echo $D | head -n1 | cut -d " " -f1`
+
+   if [[ ${D} != ${EXPECTED} ]]; then
+     echo "Directory is not where expected EXPECTED=${EXPECTED} at ${D}"
+     exit 1 
+   fi
+}
+
+try(){
+   output=$($@)
+   res=$?
+   if [[ ${res} -ne 0 ]]; then
+      exit -1
+   fi
+}
+
+install_boost() {
+   # TODO when we get better ARM support support full pinned compilations
+   # Currently Boost on aarch64-linux fails because of CMake issues and correctly detecting the toolchain
+   try apt-get install libboost-all-dev
+}
+
+install_boost
+
+
+# build Mandel
+
+echo "Building Mandel"
+pushdir ${MANDEL_DIR}
+
+try cmake -S${SCRIPT_DIR}/.. -DCMAKE_BUILD_TYPE=Release -DENABLE_OC=Off ..
```

Comment:
Is this `ENABLE_OC=Off` needed? OC should have already been disabled?
https://github.com/eosnetworkfoundation/mandel/blob/27e15012af77a2aea3cd02a1a21c5404ed32a215/CMakeLists.txt#L72-L73

More generally, is this entire script needed? What does it offer the user over just cmake && make?

Reply:
It is not necessarily needed.  Neither script is really needed we can just document what they need to do to produce the packages.

This just allows us to write up for the group who are going to produce the packages for this release.

run install_deps.h then pinned_build.sh and on ARM run install_deps.h and arm_build.sh with roughly the same inputs.

When we support this better or envelope this as part of CICD then we can remove all of these and simply the steps in the CICD recipe or Dockerfile.

Reply:
It's named arm_build.sh and literally does nothing specific for ARM afaict. It's an 80+ line script to perform the 3 commands already documented in the readme (`apt-get install`, `cmake`, `make`).

It's seemingly exactly the kind of hollow scripts we were trying to move away from.